cluster_management/src/main/java/com/pinterest/rocksplicator/Utils.java

```diff
@@ -313,12 +328,15 @@ public static long getLocalLatestSequenceNumber(String dbName, int adminPort)
   public static long getLatestSequenceNumber(String dbName, String host, int adminPort) {
     LOG.error("Get seq number from " + host + " for " + dbName);
     try {
-      Admin.Client client = getAdminClient(host, adminPort);
+      TSocket sock = new TSocket(host, adminPort);
```

Comment:
I'm wondering if it's possible to use the RAII approach here. If there is an exception in the following lines, the socket will be leaked again. Should we close the socket in the `finally` section?

Reply:
I also thought about that, and open to make that change. The reason I do it here is because: when there is any exception in creating TSocket or open the the socket, that means the socket isn't in `open` state, so there is no need to close it. 

However, agree that's not very straightward, so I'll try to use `finally`

Reply:
updated. I ended up not using `finally` only for the actual request (since it's no needer TSocket construction and open as I explained above). SO it's more like the example 4 here https://www.programcreek.com/java-api-examples/?api=org.apache.thrift.transport.TSocket. Let me know if you have other thoughts